config/application.rb

```diff
@@ -44,5 +44,8 @@ class Application < Rails::Application
 
     ManageIQ::API::Common::Logging.activate(config)
     ManageIQ::API::Common::Metrics.activate(config, "topological_inventory_api")
+
+    require Rails.root.join("app/middleware/tracker_middleware").to_s
+    config.middleware.use(::TrackerMiddleware)
```

Comment:
For tracking you probably want to make sure this is in a consistent location in the stack (probably the top of the stack.  So, you want want to use .insert instead of .use